nalgebra-macros/src/lib.rs

```diff
@@ -321,3 +322,398 @@ pub fn point(stream: TokenStream) -> TokenStream {
     };
     proc_macro::TokenStream::from(output)
 }
+
+enum ConcatElem {
+    Zero,
+    One,
+    Expr(Expr),
+}
+
+impl ConcatElem {
+    fn from_expr(expr: Expr) -> Self {
+        if let Expr::Lit(syn::ExprLit {
+            lit: Lit::Int(ilit),
+            ..
+        }) = &expr
+        {
+            if ilit.base10_digits() == "0" {
+                return ConcatElem::Zero;
+            } else if ilit.base10_digits() == "1" {
+                return ConcatElem::One;
+            }
+        }
+        ConcatElem::Expr(expr)
+    }
+
+    fn is_expr(&self) -> bool {
+        matches!(self, ConcatElem::Expr(_))
+    }
+}
+
+/// Constructs a new matrix by concatenating the given matrices.
+///
+/// **Note: Requires the `macros` feature to be enabled (enabled by default)**.
+///
+/// The syntax is similar to the [`matrix!`] and [`dmatrix!`]) macros. However the elements should
+/// be of type `&Matrix` or be one of the litterals `0` or `1`. The elements of type `&Matrix` are
```

Comment:
I just removed the identity special case code. It might not really be very useful and thinking more about it I think any sort of special syntax will look a bit weird.